### ISSUE TITLE
bug: Unable to use on NextJS v12.1.6 middleware with Edge Runtime

### ISSUE THREAD

**Issue by christopherdro** [supabase/supabase-js#461]
# Bug report

I recently tried updating to NextJS v12.1.6 and noticed that @supabase/supabase-js can no longer be used in Edge Functions. This is because native NodeJS APIs are not supported and it appears there are some dependencies of realtime-js that use these. Here is a look at the stack trace. 

## To Reproduce

Upgrade to NextJS 12.1.6 and try to use @supabase/supabase-js lib within a middleware function.

## Additional context

```
Module not found: Can't resolve 'fs'

Import trace for requested module:
./node_modules/bufferutil/index.js
./node_modules/websocket/lib/WebSocketFrame.js
./node_modules/websocket/lib/websocket.js
./node_modules/websocket/index.js
./node_modules/@supabase/realtime-js/dist/module/RealtimeClient.js
./node_modules/@supabase/realtime-js/dist/module/index.js
./node_modules/@supabase/supabase-js/dist/module/index.js
./pages/api/v0/_middleware.ts

https://nextjs.org/docs/messages/module-not-found

You're using a Node.js module (fs) which is not supported in the Edge Runtime.
Learn more: https://nextjs.org/docs/api-reference/edge-runtime
```

**Comment by christopherdro** [comment]
A temporary workaround is to use `PostgrestClient` from `@supabase/postgrest-js`

**Comment by christopherdro** [comment]
My current workaround is to use [postgrest-js](https://github.com/supabase/postgrest-js) and remove `cross-fetch` from its dependencies. 
https://github.com/christopherdro/postgrest-js/commit/ee678bc8342211f0948c9ff828d285503eb54bf5

**Comment by natterstefan** [comment]
I can confirm that we are affected by this issue as well in one of your projects.

**Comment by cagils** [comment]
We are also affected by this in middleware. Should we expect a quick fix or try to find a workaround?

**Comment by soedirgo** [comment]
I believe this PR should fix the issue: https://github.com/vercel/next.js/pull/38234

In the meantime you can try using the Node.js-based API Routes instead of the Edge API Routes - you can do this by removing the `runtime: 'experimental-edge'` config.

Additionally we're also working on changing the way we package our client libs, though this may take some time.

**Comment by soedirgo** [comment]
This should be resolved as of `next@12.2.1` :)
